### ISSUE TITLE
Feature request: table restore

### ISSUE THREAD

**Issue by dimoffon** [pgbackrest/pgbackrest#2369]
Current functionality of pgbackrest allows to restore specified databases. It would be nice to have a possibility to restore only specific set of tables from a particular database. The profits of such feature are the same as in case of databases, but the control over what should be restored is better (so, potentially we can save much space and time during recovery).
But restoring a table from the binary backup is a tricky thing. At a high-level a possible solution here can be like this:
 
1. `backup` command gathers a table metadata (in particular its `relfilenode` and `tablespace` oids) and keep it into `backup.manifest` (this logic can be triggered only with additional option to keep backward compatibility).
2. `restore` command copy only data files related to the required tables specified with new option `--table-include` (by using metadata from `backup.manifest`). In addition, it prepares file with metadata about tables that should be restored for `archive-get` command.
3. during recovery `archive-get` command will replace `XLogRecord` with a dummy one (`XLOG_NOOP`) if it is of type `RelFileNode` and its data (`spcNode`, `dbNode`, `relNode`) is absent in the metadata file (prepared during `restore` command). Once recovery is done, WAL filtering should not work anymore.

As result, recovery will end up with restoring specified tables in the database. In addition it will restore the whole system catalog of this database. 

Also we can have an alternative way of how to work with metadata. Anyway we will need to restore the system catalog. So we can gather tables' metadata right from there. As pros:

- it won't require keeping additional metadata in the repository (which may be huge);
- metadata will be prepared only if needed;
- such solution will be compatible with existing backups.

As a downside, it will require to make an intermediate recovery: first recover the system catalog only (to prepare metadata for restore command), startup and connect to the database, get the metadata, then stop the database and repeat the restore, but with tables' metadata this time.

It would be nice to know if such feature seems valuable for pgbackrest (probably it is in the backlog already). And if so, which way of the implementation is preferable or you can provide your vision on this regards.

**Comment by dwsteele** [comment]
We have discussed this feature internally before but do not have any plans to implement it. It would require a huge amount of work for what we consider to be a relatively niche case. Right now there are many more useful features that we'd prefer to implement.
